platform/windows/detect.py

```diff
@@ -654,6 +654,11 @@ def configure_mingw(env: "SConsEnvironment"):
     # https://www.scons.org/wiki/LongCmdLinesOnWin32
     env.use_windows_spawn_fix()
 
+    # In case the command line to AR is too long, use a response file.
+    env["ARCOM_ORIG"] = env["ARCOM"]
+    env["ARCOM"] = "${TEMPFILE('$ARCOM_ORIG', '$ARCOMSTR')}"
+    env["TEMPFILESUFFIX"] = ".rsp"
```

Comment:
If we want to follow SCons, we might want to do like done in their [msvc tool](https://github.com/SCons/scons/blob/master/SCons/Tool/mslib.py#L60).

For msvc it should be:
```
env['ARCOM'] = "${TEMPFILE('$AR $ARFLAGS /OUT:$TARGET $SOURCES','$ARCOMSTR')}"
```

For [all others](https://github.com/SCons/scons/blob/master/SCons/Tool/ar.py#L47) (including mingw), I guess it should be:

```
env['ARCOM'] = "${TEMPFILE('$AR $ARFLAGS $TARGET $SOURCES','$ARCOMSTR')}"
```

Reply:
That was what I originally had, but I didn't want to duplicate what had been defined in SCons itself. The approach I ended up with is similar to what the web platform is using: https://github.com/godotengine/godot/blob/7c383767a29969b0c3db036742da00ef58b765dd/platform/web/detect.py#L184-L185

Reply:
I see, I feel that also should also be changed to be honest.

That risks creating unexpected chains of TEMPFILE if the setup function is called multiple times.

If we know the toolchain, we also know the `ARCOM` format for that toolchain, so we should enforce it I think.

Reply:
Nesting `TEMPFILE` doesn't affect execution - `TempFileMunge` only creates and uses the temp file if the actual command line exceeds the set length, so only the deepest level of `TEMPFILE` takes effect, while outer ones see the already shortened command line. (I have tested it.)